From: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Date: Wed, 25 Apr 2012 01:46:18 +0000
Subject: [PATCH 45/58] be2net: Fix FW download in Lancer

commit 804c751599855a83efdc781cd9a2bda725d1391c upstream.

Increase time given by driver to adapter for completing FW download
to 30 seconds. Also return correct status when FW download times out.

Signed-off-by: Padmanabh Ratnakar <padmanabh.ratnakar@emulex.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---
 drivers/net/ethernet/emulex/benet/be_cmds.c |    8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

Index: linux-3.2.46/drivers/net/ethernet/emulex/benet/be_cmds.c
===================================================================
--- linux-3.2.46.orig/drivers/net/ethernet/emulex/benet/be_cmds.c	2013-07-26 19:21:49.000000000 +0000
+++ linux-3.2.46/drivers/net/ethernet/emulex/benet/be_cmds.c	2013-07-26 19:21:51.000000000 +0000
@@ -1824,18 +1824,16 @@
 	spin_unlock_bh(&adapter->mcc_lock);
 
 	if (!wait_for_completion_timeout(&adapter->flash_compl,
-			msecs_to_jiffies(12000)))
+					 msecs_to_jiffies(30000)))
 		status = -1;
 	else
 		status = adapter->flash_status;
 
 	resp = embedded_payload(wrb);
-	if (!status) {
+	if (!status)
 		*data_written = le32_to_cpu(resp->actual_write_len);
-	} else {
+	else
 		*addn_status = resp->additional_status;
-		status = resp->status;
-	}
 
 	return status;
 
